### PR TITLE
Add support for import statements that enable unqualified access to many syms

### PR DESCRIPTION
**Pull request by lydia-duncan**
Prior to this change you could only enable unqualified access to just a
single symbol using an import statement.  This change allows you to
list multiple symbols for unqualified access only, using `{}`.  E.g.

```chapel
module M {
  var a: int;
  proc b() { ... }
  ...
}
module User {
  import M.{a, b};
  ...
}
```

Most of the support was already present, so this change is relatively
minor.  We just add a new overload for the ImportStmt constructor and
build function, add a check to ensure the last symbol prior to the `{}`
list is always a module, and update the parser.

Adds a number of tests of this change, including:
- long lists in order of use and declaration of variables
- long lists with shuffled variables
- long lists in order of use and declaration of functions
- long lists with shuffled functions
- trying to use unqualified access on an unlisted variable (fails as desired)
- trying to use unqualified access on an unlisted function (fails as desired)
- trying to use qualified access on a listed variable (fails as desired)
- trying to use qualified access on a listed function (fails as desired)
- importing a single symbol with the curly brace syntax
- importing all the symbols in a module via two curly brace imports
- importing all the symbols in a module via a curly brace import and a single
symbol import for unqualified access
- importing all the symbols in a module in a curly brace import and enabling
qualified access via import
- Ensures the last symbol prior to the {} list is a module
- Ensures we can't have call expressions in the {} list
- Ensures we can't have literals in the {} list
- Ensures that we can mix and match imports with {} and limited use statements.
- Ensuring that we can't apply public to imports with {} just yet

Passed paratest with futures run prior to the public change, rerunning just
to be certain

### CODE REVIEW

File: compiler/AST/ImportStmt.cpp [hunk 1]
```diff
@@ -66,6 +66,32 @@ ImportStmt::ImportStmt(BaseAST* source, const char* rename,
...
+    src = new SymExpr(b);
+
+  } else if (Expr* b = toExpr(source)) {
+    src = b;
+
+  } else {
+    INT_FATAL(this, "Bad mod in ImportStmt constructor");
+  }
+
+  if (namesList->size() > 0) {
```
Comment (mppf) [inline compiler/AST/ImportStmt.cpp]: I probably wouldn't bother with this conditional (loop basically already has it)

File: compiler/AST/build.cpp [hunk 2]
```diff
@@ -567,6 +567,46 @@ BlockStmt* buildImportStmt(Expr* mod, const char* rename, bool privateImport) {
...
+          namesList.push_back(name->unresolved);
+        } else {
+          USR_FATAL(listElem->elem, "incorrect expression in 'import' for "
+                    "unqualified access, identifier expected");
+        }
+        break;
+      case PotentialRename::DOUBLE:
+        std::pair<Expr*, Expr*>* elem = listElem->renamed;
+        USR_FATAL(elem->first, "cannot rename symbols 'import'ed for "
+                  "unqualified access");
```
Comment (mppf) [inline compiler/AST/build.cpp]: I don't see a test of this (but might have missed it). Is it for the case like `import M.{x as y}` ? Should such syntax even be allowed?

Comment (lydia-duncan) [reply compiler/AST/build.cpp]: No, I meant to add one but I didn't (will do that now).  We do intend to support renaming in this case, I just didn't think doing it in this change was a good idea.

File: compiler/AST/ImportStmt.cpp (hunk 1 above)
Comment (lydia-duncan) [reply compiler/AST/ImportStmt.cpp]: I'm thinking I'll leave this, it'll be useful when renaming the symbols in the {} list is online

Comment (mppf) [reply compiler/AST/ImportStmt.cpp]: not a big deal

**Pull request merged**